patches/minecraft/net/minecraft/block/SweetBerryBushBlock.java.patch

```diff
@@ -0,0 +1,13 @@
+--- a/net/minecraft/block/SweetBerryBushBlock.java
++++ b/net/minecraft/block/SweetBerryBushBlock.java
+@@ -47,8 +47,9 @@
+    public void func_196267_b(BlockState p_196267_1_, World p_196267_2_, BlockPos p_196267_3_, Random p_196267_4_) {
+       super.func_196267_b(p_196267_1_, p_196267_2_, p_196267_3_, p_196267_4_);
+       int i = p_196267_1_.func_177229_b(field_220125_a);
+-      if (i < 3 && p_196267_4_.nextInt(5) == 0 && p_196267_2_.func_201669_a(p_196267_3_.func_177984_a(), 0) >= 9) {
++      if (i < 3 && p_196267_2_.func_201669_a(p_196267_3_.func_177984_a(), 0) >= 9 && net.minecraftforge.common.ForgeHooks.onCropsGrowPre(p_196267_2_, p_196267_3_, p_196267_1_, p_196267_4_.nextInt(5) == 0)) {
```

Comment:
Since you flipped the and, this RNG call might not be called when it was originally called unconditionally. Is that a problem?

Reply:
It is consistent with the other patches to other growing blocks, eg Cocoa:
https://github.com/Draco18s/MinecraftForge/blob/Sweet-Berry-Growth-Event/patches/minecraft/net/minecraft/block/CocoaBlock.java.patch#L7-L11

Reply:
It's done this way so that modders can override the randomness and make it happen more often.

Reply:
I think, Lex, he was worried about the having changed the side effects on the world random object's state. IMO its so minor as to be irrelevant, especially when the other growables are already altered in this fashion (I also happen to think that the vanilla ordering of those checks is wrong: random should be last).

Reply:
Ya, either way its not a issue, This is done on random ticks and thus the randomness it not very important/predictable. It doesn't change the mechanics.